File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/FrontEnd/src/app/Module/MProductValue/ProductValue.Entity.ts
import { IEntity } from "../IEntity.Entity";
import { ProductEntity } from "../MProduct/Product.Entity";
import { ProductAttributeEntity } from "../MProductAttribute/ProductAttribute.Entity";
import { LanguageEntity } from "../MLanguage/Language.Entity";

export class ProductValueEntity extends IEntity {
  Id: number = 0;
  ProductId: string = null;
  LanguageId: string = null;
  AttributeId: string = null;
  Value: string = null;
  Cx: number = null;

  AttributeEntities: ProductAttributeEntity = null;
  LanguageEntity: LanguageEntity = null;
  ProductEntity: ProductEntity = null;
  public constructor() {
    super();
  }
}

File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/Entities/ProductValueEntity.cs
using EShop.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EShop.Entities
{
    public class ProductValueEntity : BaseEntity
    {
        public Guid Id { get; set; }
        public String Value { get; set; }
        public Guid ProductId { get; set; }
        public Guid LanguageId { get; set; }
        public Guid AttributeId { get; set; }
        public ProductAttributeEntity AttributeEntity { get; set; }
        public LanguageEntity LanguageEntity { get; set; }
        public ProductEntity ProductEntity { get; set; }

        public ProductValueEntity():base() { }

        public ProductValueEntity(ProductValue ProductValue, params object[] args) :base(ProductValue)
        {
		    foreach(object arg in args)
			{
                if (arg is ProductAttribute Attribute)
                    AttributeEntity = new ProductAttributeEntity(Attribute);				
                if (arg is Language Language)
                    LanguageEntity = new LanguageEntity(Language);				
                if (arg is Product Product)
                    ProductEntity = new ProductEntity(Product);				
			}
        }
    }

    public class ProductValueSearchEntity : FilterEntity
    {
        public Guid? Id { get; set; }
        public String Value { get; set; }
        public Guid? ProductId { get; set; }
        public Guid? LanguageId { get; set; }
        public Guid? AttributeId { get; set; }
    }
}
